### ISSUE TITLE
Problem about training the model

### ISSUE THREAD

**Issue by aliinassiri** [Raman-Lab-UCLA/Multiclass_Metasurface_InverseDesign#2]
Hello, many thanks for putting these information about the model. I have problem while running the DCGAN_Train.py file. When the training loop starts after second batch ,the values of Discriminator and Generator loss don't change and D(x) ,G(D(Z)) will be 1. I don't have any idea about this problem. I run the file on windows 10, NVIDIA GEFORCE GTX 1650
![bandicam 2022-06-25 14-13-41-830](https://user-images.githubusercontent.com/107296170/175777654-a1c346f7-ba8c-4554-ac91-bebda364ac71.jpg)
.

**Comment by Orrinn** [comment]
i met the same problem, did u solve it?

**Comment by aliinassiri** [comment]
> i met the same problem, did u solve it?

i change  learning rate to 0.00001 and i think it has positive impact on the loss (i guess the problem is due to different types of GPU that we use). if you could find the problem help me about this
![2](https://user-images.githubusercontent.com/107296170/177052895-bd507b22-4a77-4bd9-87a6-65355e1fbee3.JPEG)

**Comment by Orrinn** [comment]
thank u. i gonna try it.  i tried to use different random seed, but the GAN was collapse and i got the follow loss image......
So are u GAN work properly?
![image](https://user-images.githubusercontent.com/81314168/177209891-f8551825-1d15-4eb4-9176-f476f5a58e90.png)

**Comment by aliinassiri** [comment]
> thank u. i gonna try it. i tried to use different random seed, but the GAN was collapse and i got the follow loss image...... So are u GAN work properly? ![image](https://user-images.githubusercontent.com/81314168/177209891-f8551825-1d15-4eb4-9176-f476f5a58e90.png)

sorry for replying late. I Train the model with learning rate that I have mentioned before for 500 epoch. The loss of the model is in the below image. I think because of changing  learning rate for  better result I have to train for more than 500 epoch. And other problem  I have is that when I want to validate the result with the Lumerical, the scripts that the author provided doesn't work, Do you have similar problem?
![Figure 2022-08-04 094745](https://user-images.githubusercontent.com/107296170/185790705-51d2dca9-58ef-49ac-ab4e-c007de797175.png)
.

**Comment by hzyliusha** [comment]
I doubt that the picture is not aligned with the data in excel, because the code does not consider whether the picture is aligned with the data in excel.

**Comment by Yu-Chen-Yi** [comment]
> I doubt that the picture is not aligned with the data in excel, because the code does not consider whether the picture is aligned with the data in excel.

I think there might be an issue with the 'importbinary(files{i}, 'microns');' command for loading the file. It's possible that the older version of Lumerical and the newer version have slightly different ways of importing images.
I'm going to manually input the image instead of using the command for this line. Currently, the generated absorption spectrum by Gnet doesn't differ significantly from the spectral characteristics used for training.
![360181567_626417269447776_3909386154965243711_n](https://github.com/Raman-Lab-UCLA/Multiclass_Metasurface_InverseDesign/assets/64921305/ca383400-17af-43fc-8a7f-cca90ff98794)

**Comment by xulin23** [comment]
> > 我怀疑图片与excel中的数据没有对齐，因为代码没有考虑图片与excel中的数据是否对齐。
> 
> 我认为 'importbinary(files{i}, 'microns');' 可能存在问题 用于加载文件的命令。旧版本的 Lumerical 和新版本的导入图像的方式可能略有不同。 我将手动输入图像，而不是使用此行的命令。目前，Gnet 生成的吸收光谱与用于训练的光谱特征没有显着差异。 ![360181567_626417269447776_3909386154965243711_n](https://private-user-images.githubusercontent.com/64921305/253754199-ca383400-17af-43fc-8a7f-cca90ff98794.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMwNjIwMjEsIm5iZiI6MTcwMzA2MTcyMSwicGF0aCI6Ii82NDkyMTMwNS8yNTM3NTQxOTktY2EzODM0MDAtMTdhZi00M2ZjLThhN2YtY2NhOTBmZjk4Nzk0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjIwVDA4NDIwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM5ODEzYmU2NTY0NGQ1NDg5NjdhYjEwZTY3OGVkMTUzNjc2NmM4NDBjYzYxZWRlZmY3M2RhNmI5ODA2YzllNzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0._GWKhTZs3W1mpggYMQOU1WtdWNZxAMjhTgh7w_WdXgs)

请问该如何操作

**Comment by FengNDXN** [comment]
> > thank u. i gonna try it. i tried to use different random seed, but the GAN was collapse and i got the follow loss image...... So are u GAN work properly? ![image](https://user-images.githubusercontent.com/81314168/177209891-f8551825-1d15-4eb4-9176-f476f5a58e90.png)
> 
> sorry for replying late. I Train the model with learning rate that I have mentioned before for 500 epoch. The loss of the model is in the below image. I think because of changing learning rate for better result I have to train for more than 500 epoch. And other problem I have is that when I want to validate the result with the Lumerical, the scripts that the author provided doesn't work, Do you have similar problem? ![Figure 2022-08-04 094745](https://user-images.githubusercontent.com/107296170/185790705-51d2dca9-58ef-49ac-ab4e-c007de797175.png) .

Hello, when I adjust the learning rate to 0.00001 as you said, the value of D(x) becomes zero all the time, and the discriminator can't make a judgment on the real and the output picture, do you know how to solve it?
![6d3d33efdbfb9644254595232646a63](https://github.com/Raman-Lab-UCLA/Multiclass_Metasurface_InverseDesign/assets/150985446/9c894205-731f-4854-91a0-7ff33950ff39)

**Comment by zmtttt** [comment]
"Hello, this is my loss graph. It seems that there isn't a competitive relationship forming. Could you suggest some measures for improvement?" thanks！！！！
![image](https://github.com/user-attachments/assets/0726647f-1258-4b24-b14c-1e1424bc4682)

 @Yu-Chen-Yi @xulin23 @Orrinn @hzyliusha @aliinassiri

**Comment by FengNDXN** [comment]
> “你好，这是我的损失图。似乎没有形成竞争关系。您能提出一些改进措施吗？谢谢!!! ![image](https://private-user-images.githubusercontent.com/83850606/374124683-0726647f-1258-4b24-b14c-1e1424bc4682.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg2NTA1MjIsIm5iZiI6MTcyODY1MDIyMiwicGF0aCI6Ii84Mzg1MDYwNi8zNzQxMjQ2ODMtMDcyNjY0N2YtMTI1OC00YjI0LWIxNGMtMWUxNDI0YmM0NjgyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDExVDEyMzcwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdjNmRjMGI0NGZiOWFkMGI5YWY3ODc2MTZmZjI4MjMzOTI5YzQ2YTVhY2RiN2M2MGViODRlNTYxNmQzODJiMDQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.P3S7vzCCjyxUhmAbVUPv-awfzCx1gH3TE3fvfccy960)
> 
> @Yu-Chen-Yi @xulin23 @Orrinn @hzyliusha @aliinassiri

Maybe you can change the random seed on the author's code, I changed the random seed and smaller learning rate after the loss map became look normal, but the output nanostructure picture is very bad
